lib/core/test/unit/Cardano/Wallet/DB/Properties.hs

```diff
@@ -217,6 +226,39 @@ properties = do
         it "Correctly re-construct tx history on rollbacks"
             (checkCoverage . prop_rollbackTxHistory)
 
+    describe "sparseCheckpoints" $ do
+        let mkBlk h = BlockHeader (SlotId 0 0) h (Hash "arbitrary")
+
+        it "k=2160, h=42" $ \_ -> do
+            let k = Quantity 2160
+            let h = Quantity 42
+            -- First unstable block: 0
+            sparseCheckpoints k (mkBlk h) `shouldBe`
+                [0,32,33,34,35,36,37,38,39,40,41,42]
+
+        it "k=2160, h=2414" $ \_ -> do
+            let k = Quantity 2160
+            let h = Quantity 2714
+            -- First unstable block: 554
+            sparseCheckpoints k (mkBlk h) `shouldBe`
+                [ 500  , 600  , 700  , 800  , 900
+                , 1000 , 1100 , 1200 , 1300 , 1400
+                , 1500 , 1600 , 1700 , 1800 , 1900
+                , 2000 , 2100 , 2200 , 2300 , 2400
+                , 2500 , 2600 , 2700 , 2704 , 2705
+                , 2706 , 2707 , 2708 , 2709 , 2710
+                , 2711 , 2712 , 2713 , 2714
+                ]
```

Comment:
I put some hand-written tests here to really get convinced of the behavior, in addition with the properties below.